pandablocks/utils.py

```diff
@@ -56,18 +61,23 @@ def words_to_table(
         if field_info.subtype == "int":
             # First convert from 2's complement to offset, then add in offset.
             packing_value = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
-        elif field_info.labels:
+        elif convert_enum_indices and field_info.labels:
```

Comment:
I think it's cleaner to look for `field_info.subtype == "enum"` ? I think I may have written the original check to sneakily bypass the type checking as `labels` may be None, but as this has got more complicated it's probably better to be  explicit that this `if` clause is for `enum` types only.

Reply:
Good idea